packages/fcl/src/wallet-utils/send-msg-to-fcl.js

```diff
@@ -3,8 +3,13 @@ import {onMessageFromFCL} from "./on-message-from-fcl"
 export const sendMsgToFCL = (type, msg = {}) => {
   if (window.location !== window.parent.location) {
     window.parent.postMessage({...msg, type}, "*")
-  } else {
+  } else if (window.opener) {
     window.opener.postMessage({...msg, type}, "*")
+  } else {
+    document.write("Session expired")
+    setTimeout(() => {
+      window.close()
+    }, 2500)
```

Comment:
Hoping for an opinion on this.  While we could definitely throw an error instead here it seems important that the user is aware that their wallet popup/tab is stale instead of generating a potentially unhandled error which the wallet must handle.

Reply:
`document.write` should probably be [avoided](https://developer.mozilla.org/en-US/docs/Web/API/Document/write)

This is to handle a case where parent window is closed?

Reply:
Yeah.  Specifically if the browser blocks the popup and you manually allow it afterwards, it will no longer have a reference to the opener window.  I will just replace the document.write with an error then I suppose and leave the handling up to the wallet.